Makefile

```diff
@@ -1,15 +1,51 @@
-QA_DOCKER_IMAGE=jakzal/phpqa:alpine
+QA_DOCKER_IMAGE=rollerworks/search-phpqa:latest
 QA_DOCKER_COMMAND=docker run -it --rm -v "$(shell pwd):/project" -w /project ${QA_DOCKER_IMAGE}
-ELASTICSEARCH_DOCKER_COMPOSE=lib/Elasticsearch/docker-compose.yml
 
-dist: cs-full phpstan test-full
+dist: install cs-full phpstan test-full
 ci: cs-full-check phpstan test-full
 
+install:
+	composer.phar install --no-progress --no-interaction --no-suggest --optimize-autoloader --ansi
+
+install-dev:
+	cp composer.json _composer.json
+	composer.phar config minimum-stability dev
+	composer.phar update --no-progress --no-interaction --no-suggest --optimize-autoloader --ansi
+	mv _composer.json composer.json
+
+install-lowest:
+	composer update --no-progress --no-suggest --prefer-stable --prefer-lowest --optimize-autoloader --ansi
+
 test:
 	vendor/bin/phpunit --verbose --exclude-group functional,performance
 
-test-full: es-up
+test-full:
+	export SYMFONY_DEPRECATIONS_HELPER=strict
+
+	# These tasks should be executed in paralel.
 	vendor/bin/phpunit --verbose
+	vendor/bin/phpunit --verbose --configuration travis/sqlite.travis.xml
+	vendor/bin/phpunit --verbose --configuration travis/pgsql.travis.xml
+	vendor/bin/phpunit --verbose --configuration travis/mysql.travis.xml
+
+test-with-coverage:
+	export SYMFONY_DEPRECATIONS_HELPER=strict
+
+	curl -Ls https://phar.phpunit.de/phpcov.phar > phpcov.phar
+	mkdir -p build/logs build/cov
+	phpdbg -qrr vendor/bin/phpunit --verbose --coverage-php build/cov/coverage-phpunit.cov
+	phpdbg -qrr vendor/bin/phpunit --verbose --configuration travis/sqlite.travis.xml --coverage-php build/cov/coverage-phpunit-sqlite.cov
+	phpdbg -qrr vendor/bin/phpunit --verbose --configuration travis/pgsql.travis.xml --coverage-php build/cov/coverage-phpunit-pgsql.cov
+	phpdbg -qrr vendor/bin/phpunit --verbose --configuration travis/mysql.travis.xml --coverage-php build/cov/coverage-phpunit-mysql.cov
+	phpdbg -qrr phpcov.phar merge --clover build/logs/clover.xml build/cov
+
+test-isolated: docker-up
```

Comment:
I'd go with simpler labels for targets you wish developers to use directly. For example, if you'd document the development workflow, you'd want to use simpler labels for human use, sort of like public methods vs private methods. :)